offices/api/serializers.py

```diff
@@ -0,0 +1,102 @@
+from django.core.exceptions import ValidationError
+from rest_framework.exceptions import ValidationError as DRFValidationError
+from rest_framework import serializers
+
+from offices.models import Company, Office
+
+
+class OfficeSerializer(serializers.ModelSerializer):
+    headquarter_of = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Office
+        fields = "__all__"
+
+    def validate(self, attrs):
+        for key, value in attrs.items():
+            if not key == 'company' and not value:
+                raise DRFValidationError({key: "{} can't be empty".format(key)})
+        return attrs
+
+    def get_headquarter_of(self, instance):
+        if instance.has_is_headquarter():
+            return instance.is_headquarter.id
+
+    def save(self, **kwargs):
+        try:
+            super(OfficeSerializer, self).save(**kwargs)
+        except ValidationError as e:
+            raise DRFValidationError({'company': e.messages[0]})
+
+
+class NestedOfficeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Office
+        exclude = ('company',)
+
+    def validate(self, attrs):
+        for key, value in attrs.items():
+            if not value:
+                raise DRFValidationError({key: "{} can't be empty".format(key)})
+        return attrs
+
+
+class CompanySerializer(serializers.ModelSerializer):
```

Comment:
a bit complex here, could be simplified but logically it works

Reply:
Yes, this serializer is a liitle bit complex. but it allows us to conditionally set as headquarter not only new office or existing company office but also it allows as to select office if it doesn't have company. This situation can be if we delete company. It doesn't dedele all of it's offices. So we can use this office for another company. I've tried to create FE asap so this functionality is wasn't presented on FE too